converter/src/main/java/gov/cms/qpp/conversion/encode/JsonWrapper.java

```diff
@@ -401,7 +401,12 @@ protected int validInteger(String value) {
 	 */
 	protected String validDate(String value) {
 		try {
-			LocalDate thisDate = LocalDate.parse(cleanString(value),  DateTimeFormatter.ofPattern("yyyyMMdd"));
+			String parse = cleanString(value);
+			parse = parse.replace("-", "").replace("/", "");
+			if (parse.length() > "yyyyMMdd".length()) {
+				parse = parse.substring(0, "yyyyMMdd".length());
+			}
+			LocalDate thisDate = LocalDate.parse(cleanString(parse),  DateTimeFormatter.ofPattern("yyyyMMdd"));
 			return thisDate.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
 		} catch (Exception e) {
 			throw new EncodeException(value + " is not an date of format YYYYMMDD.", e);
```

Comment:
I wonder if we should really be doing a format check on the Decode side of things so that we can return a specific error to the user to let them know that the date format they sent won't work.  It's a bit late once we get to the Encode.

Also, really the stipulation is that the date string needs to start with something that can parse into YYYY-MM-DD, it doesn't need to be exact, and can also include a time portion which we throw away.

Reply:
You may be right on the decode check, it's something we should think about.

I think this error is still appropriate given that it's what's expected of the user, everything else is just a compatibility fix. Why tell them they can pass in extra information when it's just thrown away?

Reply:
Many of these files are generated by other software and they're generating full timestamps.  If someone's upstream software is generating a timestamp in the wrong format, we should tell the user.

I'll create a new Jira issue for a date check on decode so that we can pass a useful error back to the user.  However, this will work for now to get us past the issue that's occuring.

Reply:
Sounds good to me.